### PR TITLE
Implement user agent modifying

### LINKED ISSUE #1872: Modify user agent string to indicate SA users
There has been several BaG topics describing Scratch Addons bug. To make it easier to track SA bugs, we can add subtle changes to the User Agent string added with globe button.

Additionally, we can use meta.agent attribute of project.json which is more obscure.

Example: `My browser / operating system: ChromeOS 13597.105.0, Chrome 88.0.4324.208, No Flash versions detected` (version -> version**s**)

### PR DESCRIPTION
**Pull request by apple502j**
Resolves #1872

### CODE REVIEW

File: content-scripts/cs.js [hunk 1]
```diff
@@ -17,6 +17,25 @@ if (pathArr[0] === "scratch-addons-extension") {
...
+  if (!modifyUAElem()) {
+    new MutationObserver((mutationsList, observer) => {
+      if (modifyUAElem()) {
+        observer.disconnect();
+      }
+    }).observe(document.documentElement, {
+      subtree: true,
+      childList: true,
+    });
+  }
```
[the remark this answers is not in the document]
Comment (TheColaber) [reply content-scripts/cs.js:38]: its basically a waitForElement function.

File: content-scripts/cs.js [hunk 2]
```diff
@@ -17,6 +17,25 @@ if (pathArr[0] === "scratch-addons-extension") {
 }
 if (path === "discuss/3/topic/add/") {
   window.addEventListener("load", () => forumWarning("forumWarning"));
+  let uaElemModified = false;
+  const modifyUAElem = () => {
+    if (uaElemModified) return;
+    const uaElem = document.getElementById("simple-user-agent");
+    if (uaElem) {
+      uaElem.textContent = uaElem.textContent.replace("/", "and");
```
Comment (Hans5958) [inline content-scripts/cs.js:25]: Dunno, but how about removing the surrounding spaces? I got bugged with that.
```js
uaElem.textContent = uaElem.textContent.replace(" / ", "/")
```

File: content-scripts/cs.js [hunk 3]
```diff
@@ -17,6 +17,25 @@ if (pathArr[0] === "scratch-addons-extension") {
 }
 if (path === "discuss/3/topic/add/") {
```
[the remark this answers is not in the document]
Comment (RedGuy12) [reply content-scripts/cs.js:18]: @TheColaber why resolved? its a legit question

File: content-scripts/cs.js (hunk 2 above)
Comment (RedGuy12) [reply content-scripts/cs.js:25]: i dont want it to be too suttle

File: content-scripts/cs.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (TheColaber) [reply content-scripts/cs.js:18]: bugs and glitches will only be reported in bugs and glitches.

File: content-scripts/cs.js (hunk 2 above)
Comment (WorldLanguages) [reply content-scripts/cs.js:25]: Can't we add an invisible character in a specific place? Then make cs.js or module.js detect those and put a Scratch Addons logo or something next to the user agent with a hover title that says "this is a Scratch Addons user"

**Comment by WorldLanguages** [comment]
This PR now replaces `My browser / operating system` with `My web browser / operating system`

**Comment by Hans5958** [comment]
smh just add `Scratch Addons v1.11.0`, that's what you get for doing stuff in the wrong way (and also help the mods to clear the heck out of it)

**Comment by WorldLanguages** [comment]
@Hans5958 Huh? pls explain, did you forget extension policy?

**Comment by apple502j** [comment]
@Hans5958 SA users should still be able to send a BaG report.

**Comment by Hans5958** [comment]
> 
> 
> @Hans5958 Huh? pls explain, did you forget extension policy?

it was supposed to be a joke

> 
> 
> @Hans5958 SA users should still be able to send a BaG report.

Hmm, right. Didn't want them to waste their time removing it.

**Comment by Hans5958** [comment]
Should we conduct a poll for what's going to be changed?

**Comment by WorldLanguages** [comment]
I don't see anything wrong with replacing "browser" with "web browser", so why a poll?

**Comment by WorldLanguages** [comment]
If we want to be more subtle we can also change "no Flash version detected" to "no Flash **versions** detected", unless @apple502j did not want to do that for some reason I don't know of?

**Comment by RedGuy12** [comment]
> If we want to be more subtle we can also change "no Flash version detected" to "no Flash **versions** detected", unless @apple502j did not want to do that for some reason I don't know of?

what if they have flash

**Comment by WorldLanguages** [comment]
@RedGuy12 Newer browser versions tell JS Flash is not available even when it is.

**Comment by apple502j** [comment]
Flash version is not included in simple-user-agent (which is provided server-side)

**Comment by WorldLanguages** [comment]
Ohh

**Comment by WorldLanguages** [comment]
@apple502j Can we change "version" to "versions", then? Any reason you decided to drop that idea?

**Comment by Hans5958** [comment]
> I don't see anything wrong with replacing "browser" with "web browser", so why a poll? 

What's wrong is there still suggesting coming up.

**Comment by apple502j** [comment]
@WorldLanguages ?

**Review by WorldLanguages — APPROVED** [review]

**Pull request merged**